x-pack/platform/plugins/shared/fleet/public/applications/integrations/sections/epm/screens/detail/components/back_link.tsx

```diff
@@ -6,29 +6,49 @@
  */
 
 import { EuiButtonEmpty } from '@elastic/eui';
+
 import { FormattedMessage } from '@kbn/i18n-react';
 import React, { useMemo } from 'react';
 
+import { useStartServices } from '../../../../../../../hooks';
+
 interface Props {
   queryParams: URLSearchParams;
-  href: string;
+  integrationsPath: string;
 }
 
-export function BackLink({ queryParams, href: integrationsHref }: Props) {
-  const { onboardingLink } = useMemo(() => {
+export function BackLink({ queryParams, integrationsPath }: Props) {
+  const {
+    application: { navigateToApp, getUrlForApp },
+  } = useStartServices();
+  const { returnAppId, returnPath } = useMemo(() => {
     return {
-      onboardingLink:
-        // Users from Security Solution onboarding page will have onboardingLink to redirect back to the onboarding page
-        queryParams.get('observabilityOnboardingLink') || queryParams.get('onboardingLink'),
+      // Check for custom path params to redirect back to a specified app's path
+      returnAppId: queryParams.get('returnAppId'),
+      returnPath: queryParams.get('returnPath'),
     };
   }, [queryParams]);
-  const href = onboardingLink ?? integrationsHref;
-  const message = onboardingLink ? BACK_TO_SELECTION : BACK_TO_INTEGRATIONS;
+
+  const appId = returnAppId && returnPath ? returnAppId : 'integrations';
+  const path = returnAppId && returnPath ? returnPath : integrationsPath;
+
+  const message = returnPath ? BACK_TO_SELECTION : BACK_TO_INTEGRATIONS;
 
   return (
-    <EuiButtonEmpty iconType="arrowLeft" size="xs" flush="left" href={href}>
-      {message}
-    </EuiButtonEmpty>
+    <>
+      {/* eslint-disable-next-line @elastic/eui/href-or-on-click */}
+      <EuiButtonEmpty
+        iconType="arrowLeft"
+        size="xs"
+        flush="left"
+        href={getUrlForApp(appId, { path })}
+        onClick={() => {
+          navigateToApp(appId, { path });
```

Comment:
why is there both href and onClick?

Reply:
@juliaElastic Because there was an href prior, I was hesitant to take it away

I noticed that we had that it for the Cancel as well ([link](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/index.tsx#L636-L641)). If we don't want both, I am fine to remove the href from here

Reply:
I think we can remove href then